test/model/unreads_test.dart

```diff
@@ -253,12 +254,17 @@ void main() {
       final stream = eg.stream();
       prepare();
       await store.addStream(stream);
+      await store.setUserTopic(stream, 'a', UserTopicVisibilityPolicy.muted);
+      await store.setMutedUsers([eg.thirdUser.userId]);
       fillWithMessages([
         eg.streamMessage(stream: stream, flags: []),
         eg.streamMessage(stream: stream, flags: [MessageFlag.mentioned]),
+        eg.streamMessage(stream: stream, topic: 'a', flags: [MessageFlag.mentioned]),
         eg.streamMessage(stream: stream, flags: [MessageFlag.wildcardMentioned]),
+        eg.dmMessage(from: eg.otherUser, to: [eg.selfUser], flags: [MessageFlag.mentioned]),
+        eg.dmMessage(from: eg.thirdUser, to: [eg.selfUser], flags: [MessageFlag.mentioned]),
       ]);
-      check(model.countInMentionsNarrow()).equals(2);
+      check(model.countInMentionsNarrow()).equals(4);
```

Comment:
This is probably clearest as a new test case (or two: one showing topic mutes don't matter, one showing user mutes do).

Otherwise, it rapidly gets tough to see at a glance why this result is the right one (vs 3 or 5 or 6). I think in the existing test case, it's clear enough why 2 is right: there are two messages that are mentions, and one that isn't.